### PR TITLE
[test] Use @pytest.mark.forked to fix test_without_init

### LINKED ISSUE #1406: [Test] [bug] "test_runtime.py" failing on Windows
**Describe the bug**
`ti.exe` may not work properly on Windows in some cases -- `test_runtime.py::test_without_init` (which introduced in #1347) fails on my end because it calls `ti.exe`.

**Log/Screenshots**
```
tests/python/test_runtime.py::test_without_init [Taichi] mode=development
[Taichi] <dev mode>, llvm 8.0.1, commit f2bd9829, python 3.7.4

*******************************************
**      Taichi Programming Language      **
*******************************************
C:\Users\xmk\AppData\Local\Temp\tmpqrx9npc8 is not a valid command!
usage:
    baseline            |-> Archive current benchmark result as baseline
    benchmark           |-> Run Python tests in benchmark mode
    build               |-> Build C++ files
    changelog           |-> Display changelog of current version
    debug               |-> Debug a single script
    dist                |-> Build package and test in release mode
    doc                 |-> Build documentation
    example             |-> Run an example by name (or name.py)
    format              |-> Reformat modified source files
    format_all          |-> Reformat all source files
    gif                 |-> Convert mp4 file to gif in the same directory
    regression          |-> Display benchmark regression test result
    release             |-> Make source code release
    run                 |-> Run a single script
    task                |-> Run a specific task
    test                |-> Run the tests
    update              |-> Update the Taichi codebase
    video               |-> Make a video using *.png files in the current directory
    video_crop          |-> Crop video in the same directory
    video_scale         |-> Scale video resolution in the same directory
    video_speed         |-> Speed up video in the same directory

Taichi CLI

positional arguments:
  command     command from the above list to run

optional arguments:
  -h, --help  show this help message and exit
>>> Running time: 0.00s
FAILED

=========================================== FAILURES ===========================================
______________________________________ test_without_init _______________________________________

    def test_without_init():
        # We want to check if Taichi works well without ``ti.init()``.
        # But in test ``ti.init()`` will always be called in last ``@ti.all_archs``.
        # So we have to create a new Taichi instance, i.e. test in a sandbox.
        content = '''
    import taichi as ti
    assert ti.cfg.arch == ti.cpu
    
    x = ti.var(ti.i32, (2, 3))
    assert x.shape == (2, 3)
    
    x[1, 2] = 4
    assert x[1, 2] == 4
    '''
        filename = make_temp_file()
        with open(filename, 'w') as f:
            f.write(content)
>       assert os.system(f'{sys.executable} {filename}') == 0
E       assert 1 == 0
E         -1
E         +0

tests\python\test_runtime.py:23: AssertionError
============================== 1 failed, 3 passed in 3.59 seconds ==============================
>>> Running time: 4.21s

```

**To Reproduce**
The failing test:
```python
import taichi as ti
from taichi import make_temp_file
import sys, os


def test_without_init():
    # We want to check if Taichi works well without ``ti.init()``.
    # But in test ``ti.init()`` will always be called in last ``@ti.all_archs``.
    # So we have to create a new Taichi instance, i.e. test in a sandbox.
    content = '''
import taichi as ti
assert ti.cfg.arch == ti.cpu

x = ti.var(ti.i32, (2, 3))
assert x.shape == (2, 3)

x[1, 2] = 4
assert x[1, 2] == 4
'''
    filename = make_temp_file()
    with open(filename, 'w') as f:
        f.write(content)
    assert os.system(f'{sys.executable} {filename}') == 0
```

This works:
```python
import taichi as ti
from taichi import make_temp_file
import sys, os


def test_without_init():
    # We want to check if Taichi works well without ``ti.init()``.
    # But in test ``ti.init()`` will always be called in last ``@ti.all_archs``.
    # So we have to create a new Taichi instance, i.e. test in a sandbox.
    content = '''
import taichi as ti
assert ti.cfg.arch == ti.cpu

x = ti.var(ti.i32, (2, 3))
assert x.shape == (2, 3)

x[1, 2] = 4
assert x[1, 2] == 4
'''
    filename = make_temp_file()
    with open(filename, 'w') as f:
        f.write(content)
    assert os.system(f'python -m taichi {filename}') == 0
```
(Corresponding log:)
```
tests/python/test_runtime.py::test_without_init [Taichi] mode=development
[Taichi] <dev mode>, llvm 8.0.1, commit f2bd9829, python 3.7.4

*******************************************
**      Taichi Programming Language      **
*******************************************
C:\Users\xmk\AppData\Local\Temp\tmpr6112yd1 is not a valid command!
usage:
    baseline            |-> Archive current benchmark result as baseline
    benchmark           |-> Run Python tests in benchmark mode
    build               |-> Build C++ files
    changelog           |-> Display changelog of current version
    debug               |-> Debug a single script
    dist                |-> Build package and test in release mode
    doc                 |-> Build documentation
    example             |-> Run an example by name (or name.py)
    format              |-> Reformat modified source files
    format_all          |-> Reformat all source files
    gif                 |-> Convert mp4 file to gif in the same directory
    regression          |-> Display benchmark regression test result
    release             |-> Make source code release
    run                 |-> Run a single script
    task                |-> Run a specific task
    test                |-> Run the tests
    update              |-> Update the Taichi codebase
    video               |-> Make a video using *.png files in the current directory
    video_crop          |-> Crop video in the same directory
    video_scale         |-> Scale video resolution in the same directory
    video_speed         |-> Speed up video in the same directory

Taichi CLI

positional arguments:
  command     command from the above list to run

optional arguments:
  -h, --help  show this help message and exit
>>> Running time: 0.00s
PASSED
```

**If you have local commits (e.g. compile fixes before you reproduce the bug), please make sure you first make a PR to fix the build errors and then report the bug.**

### PR DESCRIPTION
**Pull request by archibate**
Related issue = close #1406 replace #1410

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
@xumingkuan I finally found a systematic solution for this while searching doc for #1462, could you try this out?

### CODE REVIEW

**Comment by archibate** [comment]
Why AppVeyor failed test...
```
Traceback (most recent call last):
  File "C:\Users\appveyor\AppData\Roaming\Python\Python36\site-packages\py\_process\forkedfunc.py", line 110, in __del__
    if self.pid is not None:  # only clean up in main process
AttributeError: 'ForkedFunc' object has no attribute 'pid'
```
Did you meet this problem when running `ti test` or `ti test runtime`?

**Comment by xumingkuan** [comment]
> Why AppVeyor failed test...
> 
> ```
> Traceback (most recent call last):
>   File "C:\Users\appveyor\AppData\Roaming\Python\Python36\site-packages\py\_process\forkedfunc.py", line 110, in __del__
>     if self.pid is not None:  # only clean up in main process
> AttributeError: 'ForkedFunc' object has no attribute 'pid'
> ```
> 
> Did you meet this problem when running `ti test` or `ti test runtime`?

Oh, yes I did. But the test passed while showing this message when running either `ti test` or `ti test runtime`.

**Comment by archibate** [comment]
> > Why AppVeyor failed test...
> > ```
> > Traceback (most recent call last):
> >   File "C:\Users\appveyor\AppData\Roaming\Python\Python36\site-packages\py\_process\forkedfunc.py", line 110, in __del__
> >     if self.pid is not None:  # only clean up in main process
> > AttributeError: 'ForkedFunc' object has no attribute 'pid'
> > ```
> > 
> > 
> > Did you meet this problem when running `ti test` or `ti test runtime`?
> 
> Oh, yes I did. But the test passed while showing this message when running either `ti test` or `ti test runtime`.

So does it return a non-zero exit code?

**Comment by xumingkuan** [comment]
https://github.com/pytest-dev/pytest-forked/issues/6

Looks like this is not solvable with pytest?

**Comment by archibate** [comment]
Does the latest commit (xfail) works for you?

**Comment by xumingkuan** [comment]
> Does the latest commit (xfail) works for you?

The message is the same (`AttributeError: 'ForkedFunc' object has no attribute 'pid'`).

**Comment by archibate** [comment]
Are you sure? f64c864ae6f867bd0a592815ac5? Fine, let me just skip this test on win.

**Comment by xumingkuan** [comment]
> Are you sure? [f64c864](https://github.com/taichi-dev/taichi/commit/f64c864ae6f867bd0a592815ac5d125a259e33e1)? Fine, let me just skip this test on win.

Oh, this commit looks not to belong to this PR. I tried it but still `AttributeError: 'ForkedFunc' object has no attribute 'pid'`.

**Comment by xumingkuan** [comment]
> Are you sure? [f64c864](https://github.com/taichi-dev/taichi/commit/f64c864ae6f867bd0a592815ac5d125a259e33e1)? Fine, let me just skip this test on win.
> Done, [e8e9c8e](https://github.com/taichi-dev/taichi/commit/e8e9c8efdd27db39efea57920b29005f9563a78d), could you try this out?

No, it doesn't work either. However, it works if I comment out `@pytest.mark.forked`...

**Comment by archibate** [comment]
Fine, let's just move this test to `misc/`...

File: misc/test_without_init.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import taichi as ti
+import pytest
+
+
+# :
```
Comment (xumingkuan) [inline misc/test_without_init.py:5]: Remove this comment?

Comment (archibate) [reply misc/test_without_init.py:5]:
```suggestion
```

**Pull request merged**